tests/api/v2/datadog/api_logs_archives_test.go

```diff
@@ -0,0 +1,163 @@
+package test
+
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import (
+	"context"
+	"fmt"
+	"github.com/DataDog/datadog-api-client-go/api/v2/datadog"
+	"github.com/DataDog/datadog-api-client-go/tests"
+	"gopkg.in/h2non/gock.v1"
+	"reflect"
+	"testing"
+)
+
+// This test uses mocking because: 1) it relies on private data. 2) It relies on external services
+
+func TestLogsArchivesCreateS3(t *testing.T) {
+	cases := []struct {
+		archiveType string
+		action      string
+	}{
+		{
+			archiveType: "s3",
+			action:      "create",
+		},
+		{
+			archiveType: "azure",
+			action:      "create",
+		},
+		{
+			archiveType: "gcs",
+			action:      "create",
+		},
+	}
+	if tests.GetRecording() == tests.ModeReplaying {
+		t.Skip("This test case does not support reply from recording")
+	}
+	ctx, finish := WithClient(WithFakeAuth(context.Background()), t)
+	defer finish()
+	defer gock.Off()
+	client := Client(ctx)
+	assert := tests.Assert(ctx, t)
+	for _, c := range cases {
+		inputArchive := datadog.NullableLogsArchiveCreateRequest{}
+		inputArchive.UnmarshalJSON([]byte(readFixture(t, fmt.Sprintf("fixtures/logs/archives/%s/in/%s.json", c.archiveType, c.action))))
+		outputArchive := datadog.NullableLogsArchive{}
+		outputArchive.UnmarshalJSON([]byte(readFixture(t, fmt.Sprintf("fixtures/logs/archives/%s/out/%s.json", c.archiveType, c.action))))
+		URL, err := client.GetConfig().ServerURLWithContext(ctx, fmt.Sprintf("LogsArchive.%s.%s", c.action, c.archiveType))
+		assert.NoError(err)
+		gock.New(URL).Post("/api/v2/logs/config/archives").JSON(inputArchive).Reply(200).JSON(outputArchive)
+		archivesAreEqual(t, ctx, *inputArchive.Get().Data.Attributes, *outputArchive.Get().Data.Attributes)
+		assert.Equal(outputArchive.Get().Data.Type, "archives")
+		r := reflect.ValueOf(outputArchive.Get().Data.Attributes.Destination.Get().GetActualInstance())
+		assert.Equal(reflect.Indirect(r).FieldByName("Type").String(), c.archiveType)
```

Comment:
what are you trying to check here?

Reply:
You're right, there's no point on validating the deserialization of the input. I will remove it.